core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ScriptTransform.java

```diff
@@ -119,6 +119,8 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     // try evaluating the script to fail application creation if the script is invalid
     init(null);
 
+    Schema outputSchema = (schema == null) ? pipelineConfigurer.getStageConfigurer().getInputSchema() : schema;
```

Comment:
schema will always be null here as it is set in the init method. You need to parse the schema from the config (if its not null) and then do this logic.